src/query/storage/index.go

```diff
@@ -22,13 +22,21 @@ package storage
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/m3db/m3/src/dbnode/storage/index"
 	"github.com/m3db/m3/src/m3ninx/idx"
 	"github.com/m3db/m3/src/query/models"
 	"github.com/m3db/m3x/ident"
 )
 
+// QueryConvserionCache represents the query conversion LRU cache
+type QueryConvserionCache struct {
+	mu sync.RWMutex
```

Comment:
Since you're never using the Read lock, might be better to just make this a simple `sync.Mutex`